src/api.py

```diff
@@ -4,31 +4,63 @@
 from flask.ext.restful import Resource
 from flask_restful.reqparse import RequestParser
 
-from models import Reading
+from models import Reading, Device
+
+class DeviceResource(Resource):
+    """
+        The device resource handles API requests relating to robomussel data.
+	Will output all device information.
+    """
+    def get(self, site_name):
+   	"""
+            Returns all information about each device 
+   	"""
+	return map(Device.to_json, Device.query.all()) 
 
 class ReadingResource(Resource):
     """
-        The device resources handles API requests relating to robomussel data
+        The reading resources handles API requests relating to robomussel data
         it additionally supports facilities to filter datasets based on query
         string parameters
     """
-    def get(self, dev_name):
+    def get(self):
         """
             Filters the database by the given device id, and returns a JSON
             string to the requester
         """
         return map(Reading.to_json, 
-                self.filter(Reading.query.filter(Reading.device == dev_name)).all())
+                self.filter(Reading.query))
 
     def filter(self, readings):
         """
             Filters the given readings by the parameters encoded in the query
-            string
+            ()string
         """
         args = self.query_parse()
+        
```

Comment:
We can probably shorten every branch by doing 
```python
readings = readings.join(Device)
```
At the beginning, and then we can skip it on all the branches. Although that might not be ideal going forward. Your call.